src/coreclr/nativeaot/Runtime/amd64/MiscStubs.S

```diff
@@ -45,6 +45,9 @@ LOCAL_LABEL(ProbeLoop):
         ret
 NESTED_END RhpStackProbe, _TEXT
 
+// this helper is a last resort fallback for platforms that do not inline TLS access sequences
```

Comment:
Would it be possible to make this fallback to be C/C++ and C# only, so that people working on ports do not need to worry about writing any assembly?

Reply:
> Would it be possible to make this fallback to be C/C++ and C# only

I was thinking about the same thing. The reason why this is in assembly is to guarantee that the tailcall is a tailcall.